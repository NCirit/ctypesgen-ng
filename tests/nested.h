struct outer {
	struct inner {
		int x;
		int y;
	} p;
	int z;
};

struct tagged {
	enum type {
		STRING,
		INT,
	} t;
	union var {
		char *s;
		int i;
	} u;
};
