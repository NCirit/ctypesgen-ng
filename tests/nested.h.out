import ctypes

libtest = ctypes.CDLL('libtest.so')


class struct_inner(ctypes.Structure):
    pass
struct_inner._fields_ = [
    ('x', ctypes.c_int),
    ('y', ctypes.c_int),
]


class struct_outer(ctypes.Structure):
    pass
struct_outer._fields_ = [
    ('p', struct_inner),
    ('z', ctypes.c_int),
]

enum_type = ctypes.c_uint
enum_constant_STRING = 0
enum_constant_INT = 1


class union_var(ctypes.Union):
    pass
union_var._fields_ = [
    ('s', ctypes.c_char_p),
    ('i', ctypes.c_int),
]


class struct_tagged(ctypes.Structure):
    pass
struct_tagged._fields_ = [
    ('t', enum_type),
    ('u', union_var),
]
